CSV shorthand

max-heller opened the pull request:
This makes it possible to write examples in an `examples/examples.csv` file of the following format:

```csv
<PROGRAM>, <EXPECTED (optional)>
(* 1 2), 2
(- 4 2)
```

Each cell has its contents trimmed of leading and trailing whitespace.

max-heller committed:
csv shorthand

dwoos commented:
This looks great. I think it makes a lot of sense to just split on commas and say that if you *really* want to include a comma in your test you should just write it in a file. A couple things to consider:

- We could parse escaped newlines in program output. This will be useful once students can do complex output.
- We could get rid of the name field. This would let students use the CSV file as a place for "anonymous" tests, as opposed to the named tests in files. For display, we could use the program itself as the test name.

max-heller committed:
anonymous csv tests

max-heller committed:
cleanup

max-heller committed:
simplify naming of anonymous tests

max-heller commented:
> We could get rid of the name field. This would let students use the CSV file as a place for "anonymous" tests, as opposed to the named tests in files. For display, we could use the program itself as the test name.

Yeah, that makes sense. Now implemented, although I use `anonymous-0`, `anonymous-1`, etc. for naming. I added the program itself to the diff output for all cases, since it won't hurt to include.

max-heller commented:
> We could parse escaped newlines in program output. This will be useful once students can do complex output.

Not quite sure what you mean. Would this look something like the following?
```csv
(print "foo\nbar"), "foo\nbar"
```
If so, that probably already works
